src/inmanta/ast/variables.py

```diff
@@ -71,7 +71,7 @@ def execute_direct(self, requires: Dict[object, object]) -> object:
     def as_assign(self, value: ExpressionStatement, list_only: bool = False) -> AssignStatement:
         if list_only:
             raise ParserException(self.location, "+=", "Can not perform += on variable %s" % self.name)
-        return Assign(self.name, value)
+        return Assign(self, value)
```

Comment:
I don't think this is correct. `self` is of type `Reference`, not `LocatableString`. I was thinking you could make `self.name: LocatableString`, then use `str()` everywhere it's currently used. `AttributeReference` seems to violate `Reference's` type constraints though, which could start causing issues if we lose the `str()` cast in this constructor.